contrib/inventory/scaleway.py

```diff
@@ -0,0 +1,223 @@
+    ''' checks if we have a 'fresh' cache available for item requested '''
+
+    if config.has_option('cache', 'cache_dir'):
+        dpath = config.get('cache', 'cache_dir')
+
+        try:
+            existing = os.stat(
+                '/'.join([dpath, 'scaleway_ansible_inventory.json']))
+        except:
+            # cache doesn't exist or isn't accessible
+            return False
+
+        if config.has_option('cache', 'cache_max_age'):
+            maxage = config.get('cache', 'cache_max_age')
+        else:
+            maxage = 60
+        if (int(time.time()) - int(existing.st_mtime)) <= int(maxage):
+            return True
+
+    return False
+
+
+def generate_inv_from_api(config):
+    try:
+        inventory['all'] = copy.deepcopy(EMPTY_GROUP)
+
+        auth_token = env_or_param(
+            'SCALEWAY_TOKEN', config.get('auth', 'api_token'))
+        if config.has_option('compute', 'regions'):
+            regions = config.get('compute', 'regions')
+            if regions == 'all':
+                regions = ScalewayAPI.REGIONS
+            else:
+                regions = map(str.strip, regions.split(','))
+        else:
+            regions = [
+                env_or_param('SCALEWAY_REGION', fallback='par1')
+            ]
+
+        for region in regions:
+            api = ScalewayAPI(auth_token, region)
+
+            for server in api.servers():
+                hostname = server['hostname']
+                if config.has_option('defaults', 'public_ip_only') and config.getboolean('defaults', 'public_ip_only'):
+                    ip = server['public_ip']['address']
+                else:
+                    ip = server['private_ip']
+                for server_tag in server['tags']:
+                    if server_tag not in inventory:
+                        inventory[server_tag] = copy.deepcopy(EMPTY_GROUP)
+                    inventory[server_tag]['children'].append(hostname)
+                if region not in inventory:
+                    inventory[region] = copy.deepcopy(EMPTY_GROUP)
+                inventory[region]['children'].append(hostname)
+                inventory['all']['children'].append(hostname)
+                inventory[hostname] = []
+                inventory[hostname].append(ip)
+
+        return inventory
+    except Exception:
+        # Return empty hosts output
+        traceback.print_exc()
+        return {'all': {'hosts': []}, '_meta': {'hostvars': {}}}
+
+
+def get_inventory(config):
+    ''' Reads the inventory from cache or Scaleway api '''
+
+    if cache_available(config):
+        inv = get_cache('scaleway_ansible_inventory.json', config)
+    else:
+        inv = generate_inv_from_api(config)
+
+    save_cache(inv, config)
+    return json.dumps(inv)
+
+
+if __name__ == '__main__':
+    inventory = {}
+
+    # Read config
+    config = ConfigParser.SafeConfigParser()
+    for configfilename in [os.path.abspath(sys.argv[0]).rsplit('.py')[0] + '.ini', 'scaleway.ini']:
```

Comment:
just call scaleway.ini we are in ansible conflict should not occur

Reply:
@nerzhul I usually use dynamic inventories together with static inventories. As such all my inventories are in a directory, i.e:
```
├── integ/
│   ├── scaleway.ini
│   ├── scaleway.py
│   └── hosts
└── prod/
     ├── ec2.ini
     ├── ec2.py
     ├── scaleway.ini
     ├── scaleway.py
     └── hosts
```
When I call ansible it is from the root directory as such: `ansible-playbook -i prod my-playbook.yml`. This is why I need to read the `*.ini*` file from the running script.

(EC2 inventory does exactly this here: https://github.com/ansible/ansible/blob/devel/contrib/inventory/ec2.py#L308-L309)